application/src/main/java/org/togetherjava/tjbot/features/basic/MemberCountDisplayRoutine.java

```diff
@@ -0,0 +1,47 @@
+package org.togetherjava.tjbot.features.basic;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.channel.concrete.Category;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.features.Routine;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+public class MemberCountDisplayRoutine implements Routine {
+    private final Predicate<String> memberCountCategoryPredicate;
+    private static final Pattern baseNamePattern = Pattern.compile("(.+) - \\d+ Members");
+
+    public MemberCountDisplayRoutine(Config config) {
+        memberCountCategoryPredicate =
+                Pattern.compile(config.getMemberCountCategoryPattern()).asMatchPredicate();
+    }
+
+    private void updateCategoryName(Category category) {
+        int totalMemberCount = category.getGuild().getMemberCount();
+        String baseName = category.getName();
+        if (baseName.contains(" Members")) {
+            baseName = baseNamePattern.toString();
+        }
+        category.getManager()
+            .setName("%s - %d Members".formatted(baseName.trim(), totalMemberCount))
+            .queue();
+    }
+
+    @Override
+    public Schedule createSchedule() {
+        return new Schedule(ScheduleMode.FIXED_RATE, 0, 24, TimeUnit.HOURS);
+    }
+
+    @Override
+    public void runRoutine(JDA jda) {
```

Comment:
While this routine successfully finds and updates the category with the matching pattern from the config, it doesn't update it when the members count has already been appended to the category name.

Reply:
Explain further

Reply:
The diagram below should explain what's not working:

<img width="1712" alt="image" src="https://github.com/Together-Java/TJ-Bot/assets/26795353/af5db170-d0fa-495c-a087-e4e93f2e7d2a">

Reply:
fixed